### PR TITLE
Refactoring of TreeView Layout

### PR DESCRIPTION
**Pull request by Lelith**
**Who**: @lelith

**What / Why**:  Updates the layout of the TreeView and adds active state of TreeLeafs.

The centered layout was suffering from a flex-box  bug with justify-contet:center. Also there was a lot of moving of the nodes while the configuration-tree was growing, with this layout approach the important pieces stay at the same places and the nodes itself don t move as much.

storybook

### CODE REVIEW

File: lib/components/TreeView/TreeLeaf.js [hunk 1]
```diff
@@ -1,38 +1,92 @@
...
+      className,
+      { 'treeView__leaf--active': this.state.active },
+    );
+
+    return (
+      <div className={leafClasses}>
+        <div
+          className="treeView__leaf__content"
+          onKeyPress={this.toggleActive}
+          onClick={this.toggleActive}
```
Comment (cfleschhut) [inline lib/components/TreeView/TreeLeaf.js]: @Lelith While testing, noticed a glitch in the interaction

To reproduce:
- Click `TreeLeaf` to toggle `active` state to `true`
- Click `TreeLeaf` again to deactivate => all fine
- Click `TreeLeaf` to activate again
- With `TreeLeaf` being `active` click `NumberPicker` plus or minus button
=> `TreeLeaf` immediately deactivates (which makes it impossible to increase/decrease numbers by more than one without having to activate the leaf again)

**Comment by eleventhirtyfour** [comment]
my concerns here are mostly on the ux and user perception of the layout ,i don't know what you guys agreed on the last meeting but i thought the home devices would extend on the horizontal line to not break the logic of vertical depth as containers of layers  , correct me if i am wrong , the devices list will expand just vertically here right?

**Comment by Lelith** [comment]
@eleventhirtyfour after several rounds of discussion this is the version we agreed on, devices are opening vertically, entities horizontally, everything left aligned, new entities add on the right hand side.

File: lib/components/TreeView/TreeLeaf.js (hunk 1 above)
Comment (Lelith) [reply lib/components/TreeView/TreeLeaf.js]: I discovered it, too. will mount the onClick event on the Diamond, as this is also the biggest "element" of a leaf.

**Comment by eleventhirtyfour** [comment]
@Lelith  ok i was not update about the devices opening vertically , last information i had from previous meeting was just left aligned, on a side note we will save on the server  with mutations everytime there is an interaction so i guess there will be just the start button

**Pull request merged**